src/main/java/core/basesyntax/service/StorageService.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dto.Fruit;
+import core.basesyntax.dto.Operation;
+import core.basesyntax.exceptions.AlreadyHaveItException;
+
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+public class StorageService {
+    private final HashMap<String, Integer> storageMap = new HashMap<>();
+
+    public void create(Fruit fruit, Operation operation) throws AlreadyHaveItException {
+        if (storageMap.containsKey(fruit.getName())) {
+            throw new AlreadyHaveItException("This fruit already exists in storage!");
+        }
+        storageMap.put(fruit.getName(), operation.getQuantity());
+    }
+
+    public int read(Fruit fruit) {
+        if (!storageMap.containsKey(fruit.getName())) {
+            throw new NoSuchElementException("Storage does not contain this kind of fruit!");
+        }
+        return storageMap.get(fruit.getName());
+    }
+
+    public void update(Fruit fruit, Integer value) {
```

Comment:
your update and create methods look them same, do we need them both

Reply:
I use create() for balance operation logic and update for others

Reply:
bc due logic of balance operation it has appears only one time and it shows us how mny kinds of fruit we have